### PR TITLE
fix: update font

### PR DESCRIPTION
**Pull request by lyjeileen**
## Change
- Update font in vegalite (can't really tell the difference from the picture.)
- Download Inter instead of using google API to avoid font loading issue for Vegalite (I put font link in preview-head. Reference: https://storybook.js.org/docs/configure/images-and-assets#referencing-fonts-in-stories)

## Before
Screenshot 2024-05-24 at 11 38 44 AM

## After
Screenshot 2024-05-24 at 11 38 05 AM

### CODE REVIEW

File: src/components/chart/vegaLiteChart/vegaLiteChart.css [hunk 1]
```diff
@@ -6,3 +6,9 @@
 .rustic-vega-lite-chart {
   flex: 1;
 }
+
+#vg-tooltip-element.vg-tooltip {
+  border-radius: 16px;
+  box-shadow: none;
+  font-family: 'Inter';
```
Comment (kaseyvee) [inline src/components/chart/vegaLiteChart/vegaLiteChart.css:13]: Is font-family here still needed?

**Review by Shiti — CHANGES_REQUESTED** [review]
I would recommend that instead of hard-coding font as `Inter`, we should refer it to from the theme. Get the value from `useTheme()`

File: src/components/chart/vegaLiteChart/vegaLiteChart.tsx [hunk 2]
```diff
@@ -32,7 +66,30 @@ function VegaLiteChart(props: VegaLiteChartData) {
...
+    function getFont() {
+      if (props.options?.config) {
+        if (typeof props.options.config === 'string') {
+          return props.options.config
+        } else {
+          return (props.options.config.font as string) || 'Inter'
+        }
+      } else {
+        return 'Inter'
+      }
```
Comment (Shiti) [inline src/components/chart/vegaLiteChart/vegaLiteChart.tsx:89]:
```suggestion
      (props.options?.config && typeof props.options.config === 'string') ? props.options.config : deafultFont
```

File: src/components/visualization/vegaLiteViz/vegaLiteViz.tsx [hunk 3]
```diff
@@ -12,26 +13,50 @@ import type { VegaLiteData } from '../../types'
...
+        } else {
+          combinedConfig = {
+            ...defaultConfig,
+            ...config,
+            font: config.font || defaultConfig.font,
+          }
         }
+      } else {
+        combinedConfig = defaultConfig
+      }
```
Comment (Shiti) [inline src/components/visualization/vegaLiteViz/vegaLiteViz.tsx:41]: could this be simplified ? maybe something like this -
```suggestion
      if (props.options?.config && typeof props.options.config !== 'string'){
         combinedConfig = {...props.options.config, ...defaultFontConfig}
       }
```

**Review by Shiti — CHANGES_REQUESTED** [review]

File: src/components/visualization/vegaLiteViz/vegaLiteViz.tsx (hunk 3 above)
Comment (lyjeileen) [reply src/components/visualization/vegaLiteViz/vegaLiteViz.tsx]: I also wants to set default font when there's no config/options provided and this method won't cover that scenario. I've pushed my changes. Let me know if you have any feedback

File: src/components/visualization/vegaLiteViz/vegaLiteViz.tsx [hunk 4]
```diff
@@ -12,26 +13,35 @@ import type { VegaLiteData } from '../../types'
...
-  useEffect(() => {
-    function renderChart() {
-      if (chartRef.current && props.spec) {
-        const options = {
-          ...props.options,
-          theme: isDarkTheme ? props.theme?.dark : props.theme?.light,
-        }
+  function renderChart() {
+    if (chartRef.current && props.spec) {
+      const { config, ...restOptions } = props.options || {}
```
Comment (Shiti) [inline src/components/visualization/vegaLiteViz/vegaLiteViz.tsx:22]: would this work ? 
```suggestion
      let options = {
        ...props.options, 
         theme: isDarkTheme ? props.theme?.dark : props.theme?.light
      }   
      if (!props.options?.config?.font) {
          options.config.font = defaultFont
      }
```

**Review by Shiti — APPROVED** [review]

**Pull request merged**